pages/signup.tsx

```diff
@@ -129,13 +131,26 @@ const SignupForm: React.FC<SignupFormProps> = ({
 
 const Signup: React.FC = () => {
   const [signupSuccess, setSignupSuccess] = useState(false)
-  const [signupErrors, setSignupErrors] = useState({} as SignupErrors)
+  const [signupErrors, setSignupErrors] = useState({
+    confirmEmail: false,
+    userName: false
+  } as SignupErrors)
+  const [signupUser] = useMutation(SIGNUP_USER)
   const handleSubmit = async (values: Values) => {
-    const data = await signupUser(values)
-    if (data.success) {
-      setSignupSuccess(true)
-    } else {
-      setSignupErrors(Object.assign({}, data.errorMessage))
+    try {
+      const { data } = await signupUser({ variables: values })
+      if (data.signup.success) {
+        setSignupSuccess(true)
+      }
+    } catch (error) {
+      const newSignupErrors = { ...signupErrors }
+      const message = _.get(error, 'message', '')
+      if (message) {
+        message.indexOf('Email') !== -1
+          ? (newSignupErrors.confirmEmail = true)
+          : (newSignupErrors.userName = true)
+      }
+      setSignupErrors(Object.assign({}, newSignupErrors))
```

Comment:
This code confuse me...
we pass an object `error` to `_.get(error, 'message', '')` that will check if we have the key `Email`
if `false` that will return an empty `string`
if `true` that will return an `array`
then we mutate the `newSignupErrors` object and we update the state with a new object again
Do I have correct?
I feel it really confusing...

Reply:
yeah so when you catch the error it is a ApolloClient error object and the error is thrown here as a UserInputError:

https://github.com/garageScript/c0d3-app/blob/master/helpers/controllers/authController.ts#L110-L122 

So then when I do ```_.get(error, 'message', '')``` It will give me the error message string or empty string. Then I check if message exists and if it does then I see if the message has the substring ```Email``` in it meaning that the error message is about the email otherwise if ```message.indexOf('Email')``` returns -1 then it would be the ```User already Exists!``` error message.

Then based on that I update the signupErrors and update the state, where the signup errors are passed into the SignupForm component right below the above code 
```<SignupForm handleSubmit={handleSubmit} signupErrors={signupErrors} />```

If there is still any confusion or if you if a better implementation idea let me know!

Reply:
If message just a string maybe ```.contains``` would work better for understanding the code.

Is it not possible to have multiple Errors? Seems like the backend is working to only throw one error right now, not sure if we want to aggregate errors and display multiple.

Reply:
Instead of doing this, why don't we just `useEffect` to check if error exists. 

It seems way simpler than using this logic. There's no need to write our own logic where `ApolloClient` provides this.